### ISSUE TITLE
Vue reactivity

### ISSUE THREAD

**Issue by Tordek** [wilm0x42/wVote#77]
The reason we needed the "updateNumEntriesRated" hack is due to a limitation of Vue's reactivity: Vue cannot detect properties being added to an object after it's left `data()`, but votes might be added in the `entry` object later than that.

E.g.: The `vote` page queries for Entry data, and receives week information like this:

```
    {
        ...
        entries: [{
            entrantName: "Tordek",
            entryName: "Test",
            mp3Format: "mp3",
            mp3Url: "http://127.0.0.1",
            pdfUrl: "http://127.0.0.1",
            uuid: "123",
        }],
       voteParams: ['prompt', 'score', 'overall'],
       ...
    }
```

Note how the entry doesn't have `prompt`, `score` or `overall` paramters.

Later, when a `star-voting-widget` is generated for that entry's `prompt` property, a value is assigned to it, but it does not trigger a reactivity update.

The best solution is probably to guarantee server-side that these properties always exist on the object (even if the value is `null`), to aid on the changes for #42

**Comment by wilm0x42** [comment]
We do already have this bit in `http_server.py` line 454
```py
        # dummy vote data for the client's benefit
        for voteParam in week["voteParams"]:
            prunedEntry[voteParam] = None
```
Would we just need to set that to `0` instead of `None`, or am I thinking of the wrong code path entirely?

**Comment by Tordek** [comment]
I was sure there was a time where we had trouble syncing the voting widgets, either the individual one with the list, or the voting with the vote count, and that's why we ended up on the `updateNumEntriesRated` version.

It might have been fixed when I removed state from the star widget and I missed it.

In any case, this allows removing `numEntriesRated` from `data`, converting it to a calculated property, and removing the `@input="updateNumEntriesRated"` triggers.
